django_cloud_tasks/tasks/task.py

```diff
@@ -218,23 +219,33 @@ def asap(cls, **kwargs):
 
     @classmethod
     def later(cls, task_kwargs: dict, eta: int | timedelta | datetime, queue: str = None, headers: dict | None = None):
+        delay_in_seconds = cls._calculate_delay_in_seconds(eta)
+        cls._validate_delay(delay_in_seconds)
+        return cls.push(
+            task_kwargs=task_kwargs,
+            queue=queue,
+            headers=headers,
+            delay_in_seconds=delay_in_seconds,
+        )
+
+    @staticmethod
+    def _calculate_delay_in_seconds(eta: int | timedelta | datetime) -> int:
         if isinstance(eta, int):
-            delay_in_seconds = eta
+            return eta
         elif isinstance(eta, timedelta):
-            delay_in_seconds = eta.total_seconds()
+            return int(eta.total_seconds())
         elif isinstance(eta, datetime):
-            delay_in_seconds = (eta - now()).total_seconds()
+            return int((eta - now()).total_seconds())
         else:
             raise ValueError(
-                f"Unsupported schedule {eta} of type {eta.__class__.__name__}. " "Must be int, timedelta or datetime."
+                f"Unsupported schedule {eta} of type {eta.__class__.__name__}. Must be int, timedelta or datetime."
             )
 
-        return cls.push(
-            task_kwargs=task_kwargs,
-            queue=queue,
-            headers=headers,
-            delay_in_seconds=delay_in_seconds,
-        )
+    @staticmethod
+    def _validate_delay(delay_in_seconds: int):
+        max_eta_task = getattr(settings, "MAXIMUM_ETA_TASK", None)
```

Comment:
We must add some [prefix](https://github.com/flamingo-run/django-cloud-tasks/blob/main/django_cloud_tasks/apps.py#L13) because this is a reusable django-app, which means that it'll be used on top of other Django apps.
These apps might already use the setting `MAXIMUM_ETA_TASK`, leading to unwanted usage by this liberary.

Reply:
done